TSPL.docc/GuidedTour/GuidedTour.md

```diff
@@ -2257,6 +2257,158 @@ print(fridgeIsOpen)
   ```
 -->
 
+## Concurrency
+
+You use `async` to mark an asynchronous function.
+You mark an asynchronous operation,
+like a call to an asynchronous function,
+by writing `await`
+if you want to wait for it to complete.
+
+```swift
+func bake(_ food: String) async -> String {
+    // ... wait for food to bake ...
+    return food
+}
+func makeCookies() async -> String {
+    let cookies = await bake("cookies")
+    return cookies
+}
+```
+
+<!--
+  - test: `guided-tour`
+
+  ```swifttest
+  -> func bake(_ food: String) async -> String {
+         // ... wait for food to bake ...
+         return food
+     }
+  -> func makeCookies() async -> String {
+         let cookies = await bake("cookies")
+         return cookies
+     }
+  ```
+-->
+
+You can use `async`-`let` to start an asynchronous operation
+without waiting for it to complete,
+which lets the operations run at the same time.
+Because you still need to wait for the operation to finish
+before you can use the value it returns,
+you write `await` when you read that constant's value.
+
+```swift
+async let cookies = makeCookies()
+async let bread = bake("bread")
+
+let bakedGoods = await [cookies, bread]
+```
+
+<!--
+  - test: guided-tour-async
+
+  ```swifttest
+  -> async let cookies = makeCookies()
+  -> async let bread = bake("bread")
+  ---
+  -> let bakedGoods = await [cookies, bread]
+  ```
+-->
+
+You structure concurrent code using task groups and child tasks.
+Writing `async`-`let` implicitly creates a child task.
+
+```swift
+let desserts = await withTaskGroup(of: String.self) { group in
+    for recipe in ["biscuit", "croissant", "egg tart"] {
+        group.addTask {
+            return await bake(recipe)
+        }
+    }
+
+    var results: [String] = []
+    for await result in group {
+        results.append(result)
+    }
+
+    return results
+}
+```
+
+
+<!--
+  - test: guided-tour-async
+
+  ``` swifttest
+  -> let desserts = await withTaskGroup(of: String.self) { group in
+         for recipe in ["biscuit", "croissant", "egg tart"] {
+             group.addTask {
+                 return await bake(recipe)
+             }
+         }
+  ---
+         var results: [String] = []
+         for await result in group {
+             results.append(result)
+         }
+  ---
+         return results
+     }
+  >> print(desserts)
+  << ["biscuit", "croissant", "egg tart"]
+  ```
+-->
+
+Actors are similar to classes,
+except they ensure that different asynchronous functions
+can all interact with an instance of the same actor at the same time.
```

Comment:
"... at the same time, safely."

Technically people could call methods on classes from different dispatch queues and threads before, but then they'd run into  data races.

Reply:
Yes, this is worth pointing out.

Changed in b2bbaaca.